packages/@aws-cdk/aws-codedeploy/test/lambda/test.custom-deployment-config.ts

```diff
@@ -0,0 +1,301 @@
+            timeBasedCanary: {
+              canaryPercentage: '5',
+              canaryInterval: '1',
+            },
+            type: 'TimeBasedCanary',
+          },
+        },
+        physicalResourceId: {
+          id: 'MyDeploymentConfig',
+        },
+      },
+      Delete: {
+        action: 'deleteDeploymentConfig',
+        service: 'CodeDeploy',
+        parameters: {
+          deploymentConfigName: 'MyDeploymentConfig',
+        },
+      },
+    }));
+
+    expect(stack).to(haveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'codedeploy:CreateDeploymentConfig',
+            Effect: 'Allow',
+            Resource: '*',
+          },
+          {
+            Action: 'codedeploy:DeleteDeploymentConfig',
+            Effect: 'Allow',
+            Resource: '*',
+          },
+        ],
+        Version: '2012-10-17',
+      },
+    }));
+    test.done();
+  },
+  'can create linear custom config'(test: Test) {
+
+    // WHEN
+    const config = new codedeploy.CustomLambdaDeploymentConfig(stack, 'CustomConfig', {
+      type: codedeploy.CustomLambdaDeploymentConfigType.LINEAR,
+      interval: cdk.Duration.minutes(1),
+      percentage: 5,
+    });
+    new codedeploy.LambdaDeploymentGroup(stack, 'MyDG', {
+      application,
+      alias,
+      deploymentConfig: config,
+    });
+
+    // THEN
+    expect(stack).to(haveResource('AWS::CodeDeploy::DeploymentGroup', {
+      DeploymentConfigName: 'CustomConfig.LambdaLinear5PercentEvery1Minutes',
+      DeploymentStyle: {
+        DeploymentOption: 'WITH_TRAFFIC_CONTROL',
+        DeploymentType: 'BLUE_GREEN',
+      },
+    }));
+
+    test.done();
+  },
+  'can create canary custom config'(test: Test) {
+
+    // WHEN
+    const config = new codedeploy.CustomLambdaDeploymentConfig(stack, 'CustomConfig', {
+      type: codedeploy.CustomLambdaDeploymentConfigType.CANARY,
+      interval: cdk.Duration.minutes(1),
+      percentage: 5,
+    });
+    new codedeploy.LambdaDeploymentGroup(stack, 'MyDG', {
+      application,
+      alias,
+      deploymentConfig: config,
+    });
+
+    // THEN
+    expect(stack).to(haveResource('AWS::CodeDeploy::DeploymentGroup', {
+      DeploymentConfigName: 'CustomConfig.LambdaCanary5Percent1Minutes',
+      DeploymentStyle: {
+        DeploymentOption: 'WITH_TRAFFIC_CONTROL',
+        DeploymentType: 'BLUE_GREEN',
+      },
+    }));
+    test.done();
+  },
+  'dependency on the config exists to ensure ordering'(test: Test) {
+
+    // WHEN
+    const config = new codedeploy.CustomLambdaDeploymentConfig(stack, 'CustomConfig', {
+      type: codedeploy.CustomLambdaDeploymentConfigType.CANARY,
+      interval: cdk.Duration.minutes(1),
+      percentage: 5,
+    });
+    new codedeploy.LambdaDeploymentGroup(stack, 'MyDG', {
+      application,
+      alias,
+      deploymentConfig: config,
+    });
+
+    // THEN
+    expect(stack).to(haveResource('AWS::CodeDeploy::DeploymentGroup', {
```

Comment:
Same here - can we use `haveResourceLike` instead, and just focus on the important properties, not all of them?